include/gz/sensors/AirFlowSensor.hh

```diff
@@ -0,0 +1,101 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_SENSORS_AIRFLOWSENSOR_HH_
+#define GZ_SENSORS_AIRFLOWSENSOR_HH_
+
+#include <memory>
+
+#include <sdf/sdf.hh>
+
+#include <gz/utils/SuppressWarning.hh>
+
+#include <gz/sensors/config.hh>
+#include <gz/sensors/air_flow/Export.hh>
+
+#include "gz/sensors/Sensor.hh"
+
+namespace gz
+{
+  namespace sensors
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace GZ_SENSORS_VERSION_NAMESPACE {
+    //
+    /// \brief forward declarations
+    class AirFlowSensorPrivate;
+
+    /// \brief AirFlow Sensor Class
+    ///
+    /// A sensor that reports air speed through differential pressure readings.
+    class GZ_SENSORS_AIR_FLOW_VISIBLE AirFlowSensor :
+      public Sensor
+    {
+      /// \brief constructor
+      public: AirFlowSensor();
+
+      /// \brief destructor
+      public: virtual ~AirFlowSensor();
+
+      /// \brief Load the sensor based on data from an sdf::Sensor object.
+      /// \param[in] _sdf SDF Sensor parameters.
+      /// \return true if loading was successful
+      public: virtual bool Load(const sdf::Sensor &_sdf) override;
+
+      /// \brief Load the sensor with SDF parameters.
+      /// \param[in] _sdf SDF Sensor parameters.
+      /// \return true if loading was successful
+      public: virtual bool Load(sdf::ElementPtr _sdf) override;
+
+      /// \brief Initialize values in the sensor
+      /// \return True on success
+      public: virtual bool Init() override;
+
+      /// \brief Get the current velocity.
+      /// \return Current velocity of the sensor.
+      public: gz::math::Vector3d Velocity() const;
+
+      /// \brief Update the velocity of the sensor
+      /// \param[in] _vel The velocity of the sensor [m/s]
+      public: void SetVelocity(const gz::math::Vector3d &_vel);
+
+      /// \brief Update the wind velocity
+      /// \param[in] _vel The wind velocity [m/s]
+      public: void SetWindVelocity(const gz::math::Vector3d &_vel);
+
+      using Sensor::Update;
+
+      /// \brief Update the sensor and generate data
+      /// \param[in] _now The current time
+      /// \return true if the update was successfull
+      public: virtual bool Update(
+        const std::chrono::steady_clock::duration &_now) override;
+
+      /// \brief Check if there are any subscribers
+      /// \return True if there are subscribers, false otherwise
+      public: virtual bool HasConnections() const override;
+
+      GZ_UTILS_WARN_IGNORE__DLL_INTERFACE_MISSING
+      /// \brief Data pointer for private data
+      /// \internal
+      private: std::unique_ptr<AirFlowSensorPrivate> dataPtr;
+      GZ_UTILS_WARN_RESUME__DLL_INTERFACE_MISSING
+    };
```

Comment:
Can you use `GZ_UTILS_IMPL_PTR` instead? (see https://github.com/gazebosim/gz-sensors/blob/ec52913e608ce8f64053650bfcced84a70ca12e2/include/gz/sensors/Distortion.hh#L103)